cgen error in auto generated str() method of an enum with duplicate enum keys

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.6 a7db095.22df56f
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /space/v/oo/v
vexe mtime: 2024-07-23 06:24:44

vroot: OK, value: /space/v/oo
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.45.2
Git vroot status: weekly.2024.30-2-g22df56fa (1 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 a0799a5b

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg b.v`
```v
const one = 1

enum MyEnum {
	a = one
	b
	c = one
}

$for val in MyEnum.values {
	println('> name: ${val.name} | value: ${val.value}')
}
```

**What did you expect to see?**

a checker error, that `a` and `c` are duplicates

**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/b.01J3F8XFC74K61874J8G3H137K.tmp.c:2172: error: duplicate case value
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1951,6 +1951,14 @@ fn (mut c Checker) enum_decl(mut node ast.EnumDecl) {
 						}
 						if field.expr.kind == .constant && field.expr.obj.typ.is_int() {
 							// accepts int constants as enum value
+							if mut field.expr.obj is ast.ConstField {
+								if mut field.expr.obj.expr is ast.IntegerLiteral {
+									c.check_enum_field_integer_literal(field.expr.obj.expr,
+										signed, node.is_multi_allowed, senum_type, field.expr.pos, mut
+										useen, enum_umin, enum_umax, mut iseen, enum_imin,
+										enum_imax)
+								}
+							}
 							continue
 						}
 					}

--- a/vlib/v/checker/tests/enum_field_value_duplicate_d.out
+++ b/vlib/v/checker/tests/enum_field_value_duplicate_d.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/enum_field_value_duplicate_d.vv:6:6: error: enum value `1` already exists
+    4 |     a = one
+    5 |     b
+    6 |     c = one
+      |         ~~~
+    7 | }
+    8 |

--- a/vlib/v/checker/tests/enum_field_value_duplicate_d.vv
+++ b/vlib/v/checker/tests/enum_field_value_duplicate_d.vv
@@ -1,0 +1,13 @@
+const one = 1
+
+enum MyEnum {
+	a = one
+	b
+	c = one
+}
+
+fn main() {
+	$for val in MyEnum.values {
+		println('> name: ${val.name} | value: ${val.value}')
+	}
+}

